.ai-doc/50-migration-angular/10-migration-diffs/component-diffs.md

```diff
@@ -0,0 +1,540 @@
+        on-click="$ctrl.edit($row)"
+      ></oui-action-menu-item>
+      <oui-action-menu-item
+        text="{{'delete' | translate}}"
+        on-click="$ctrl.delete($row)"
+      ></oui-action-menu-item>
+    </oui-action-menu>
+  </oui-column>
+</oui-datagrid>
+```
+
+### React Equivalent
+
+```typescript
+import { Datagrid, DatagridColumn } from '@ovh-ux/manager-react-components';
+import { OdsButton, OdsPopover } from '@ovhcloud/ods-components/react';
+
+type Service = {
+  serviceName: string;
+  status: string;
+};
+
+const columns: DatagridColumn<Service>[] = [
+  {
+    id: 'serviceName',
+    label: t('name'),
+    cell: (row) => row.serviceName,
+    isSortable: true,
+  },
+  {
+    id: 'status',
+    label: t('status'),
+    cell: (row) => t(row.status),
+  },
+  {
+    id: 'actions',
+    label: t('actions'),
+    cell: (row) => (
+      <OdsPopover>
+        <OdsButton slot="trigger" variant="ghost" icon="ellipsis-vertical" />
+        <OdsButton
+          variant="ghost"
+          label={t('edit')}
+          onClick={() => handleEdit(row)}
+        />
+        <OdsButton
+          variant="ghost"
+          label={t('delete')}
+          onClick={() => handleDelete(row)}
+        />
+      </OdsPopover>
+    ),
+  },
+];
+
+return (
+  <Datagrid
+    columns={columns}
+    items={services}
+    pageSize={10}
+    pageSizeOptions={[10, 25, 50]}
+  />
+);
+```
+
+### Transformation Rules
+
+1. `<oui-datagrid rows="...">` → `<Datagrid items={...} />`
+2. `<oui-column>` → Column definition object in array
+3. `property="x"` → `cell: (row) => row.x`
+4. `sortable` → `isSortable: true`
+5. `title="'text'"` → `label: t('key')`
+6. `ng-bind="... | translate"` → `t(value)` in cell function
+7. `<oui-action-menu>` → `<OdsPopover>` with buttons
+
+---
+
+## Datagrid with Server-side Pagination
+
+### AngularJS
+
+```html
+<oui-datagrid
+  rows-loader="$ctrl.loadRows($config)"
+  page-size="25"
+>
+  <!-- columns -->
+</oui-datagrid>
+```
+
+```javascript
+// Controller
+$ctrl.loadRows = (config) => {
+  return iceberg('/dedicated/nasha')
+    .query()
+    .expand('CachedObjectList-Pages')
+    .limit(config.pageSize)
+    .offset(config.offset)
+    .sort(config.sort.property, config.sort.dir === 1 ? 'ASC' : 'DESC')
+    .execute()
+    .$promise;
+};
+```
+
+### React Equivalent
+
+```typescript
+import { useDataApi } from '@ovh-ux/manager-react-components';
+
+const ListingPage = () => {
+  const {
+    data,
+    isLoading,
+    pageIndex,
+    pageSize,
+    setPageIndex,
+    setPageSize,
+    sorting,
+    setSorting,
+  } = useDataApi({
+    route: '/dedicated/nasha',
+    queryKey: ['nasha', 'list'],
+  });
+
+  return (
+    <Datagrid
+      columns={columns}
+      items={data ?? []}
+      pageSize={pageSize}
+      pageIndex={pageIndex}
+      onPageChange={setPageIndex}
+      onPageSizeChange={setPageSize}
+      sorting={sorting}
+      onSortChange={setSorting}
+      isLoading={isLoading}
+    />
+  );
+};
+```
```

Comment:
It's not the same, for loadRows behaviour we need to create a child components for cell that depend on the secondary API calls.